src/main/target/common_deprecated_post.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Flag any deprecated defines with compile errors so they
+// can be cleaned up and not further propagated.
+
+#pragma once
+
+#ifdef ACC_1_ALIGN
+#error "The ACC_1_ALIGN define has been deprecated - please remove from the target definition"
```

Comment:
I am wondering if this should only issue warnings - this way it won't be a hindrance for developers working on porting targets, but it will crop up immediately once it is submitted as a pull request.

Reply:
I can't really think of any scenario where it's really much of a hindrance. If you're working on a new target and for some reason you base it on another target but use the one from a previous version then you probably made a mistake. Probably the most consternation will come from manufacturers that are hosting custom targets (probably violating GPL) when they try to rebase for a new release. But the fixes are really trivial and clearly explained. The problem with only issuing warnings is that people will ignore them and still submit PR's causing more churn for us. Yes I realize that CI will fail the job but I'd rather have force them to have a working target before submitting the PR.

Reply:
as noted in slack:

I'm in the middle of rebasing code that has a new target.
"But the fixes are really trivial and clearly explained" - sure, I can delete the lines.
but it doesn't explain where the settings went or if there are any new settings that should be made instead.
e.g. for `ACC_1_ALIGN`...
so, my two cents, I'm left trawling the git commit history and closed PR's trying to find what happened to the settings, likely everyone else doing this will be in the same boat.

Reply:
Also as noted in the slack: This PR didn't cause the defines to be deprecated - it's just about preventing them from reappearing in targets and alerting that they're no longer used. The previous state was that there was no notification and the developer had no indication that the settings they thought they were making had no effect so I'm not sure how that's better.